contracts/liquidity_hub/pool-manager/src/tests/integration_tests.rs

```diff
@@ -171,191 +154,77 @@ fn deposit_and_withdraw_sanity_check() {
-        },
-    ];
-
-    // Default Pool fees white_whale_std::pool_network::pair::PoolFee
-    #[cfg(not(feature = "osmosis"))]
-    let pool_fees = PoolFee {
-        protocol_fee: Fee {
-            share: Decimal::zero(),
-        },
-        swap_fee: Fee {
-            share: Decimal::zero(),
-        },
-        burn_fee: Fee {
-            share: Decimal::zero(),
-        },
-    };
 
-    #[cfg(feature = "osmosis")]
-    let pool_fees = PoolFee {
-        protocol_fee: Fee {
-            share: Decimal::zero(),
-        },
-        swap_fee: Fee {
-            share: Decimal::zero(),
-        },
-        burn_fee: Fee {
-            share: Decimal::zero(),
-        },
-        osmosis_fee: Fee {
-            share: Decimal::zero(),
-        },
-    };
-
-    // Create a pair
-    suite
-        .instantiate_with_cw20_lp_token()
-        .add_native_token_decimals(creator.clone(), "uwhale".to_string(), 6)
-        .create_pair(
-            creator.clone(),
-            asset_infos,
-            pool_fees,
-            white_whale_std::pool_network::asset::PairType::ConstantProduct,
-            false,
-            None,
-            vec![coin(1000, "uusd")],
-            |result| {
-                result.unwrap();
-            },
-        );
-    suite.increase_allowance(
+    // Lets try to withdraw liquidity
+    suite.withdraw_liquidity(
         creator.clone(),
-        suite.cw20_tokens[0].clone(),
-        Uint128::from(1000000u128),
-        suite.pool_manager_addr.clone(),
-    );
-    // Lets try to add liquidity
-    suite.provide_liquidity(
-        creator.clone(),
-        "0".to_string(),
+        "whale-uluna".to_string(),
         vec![
-            Asset {
-                info: AssetInfo::NativeToken {
-                    denom: "uwhale".to_string(),
-                },
-                amount: Uint128::from(1000000u128),
+            Coin {
+                denom: "uwhale".to_string(),
+                amount: Uint128::from(1_000_000u128),
             },
-            Asset {
-                info: AssetInfo::Token {
-                    contract_addr: suite.cw20_tokens[0].to_string(),
-                },
-                amount: Uint128::from(1000000u128),
+            Coin {
+                denom: "uluna".to_string(),
+                amount: Uint128::from(1_000_000u128),
             },
         ],
         vec![Coin {
-            denom: "uwhale".to_string(),
-            amount: Uint128::from(1000000u128),
+            denom: "whale-uluna".to_string(),
+            amount: Uint128::from(1_000_000u128) - MINIMUM_LIQUIDITY_AMOUNT,
         }],
         |result| {
-            // Ensure we got 999000 in the response which is 1mil less the initial liquidity amount
-            for event in result.unwrap().events {
-                println!("{:?}", event);
-            }
+            // Ensure we got 999_000 in the response which is 1_000_000 less the initial liquidity amount
```

Comment:
I think this comment is slightly wrong -- we are trading 999k shares for 1mill of our liquidity.

Reply:
+1